lib/esextractor.cpp

```diff
@@ -100,6 +100,12 @@ ESExtractor::prepare (const char *uri, const char *options)
   return found;
 }
 
+void ESExtractor::setOptions (const char *options)
+{
+    if (m_stream)
+      return m_stream->setOptions (options);
```

Comment:
I'm not sure about this API. It looks racy if somebody call it from another thread.

And I wouldn't use `if (m_stream)` but rather `assert (m_stream)` since it's supposed to be created at esextractor instantiation (if using the C API... and we should remove the C++ API.. but that's another bug)